core/common/src/main/java/alluxio/util/CommonUtils.java

```diff
@@ -282,10 +284,25 @@ public static void warmUpLoop() {
    * @return the groups list that the {@code user} belongs to. The primary group is returned first
    */
   public static List<String> getUnixGroups(String user) throws IOException {
+    return getUnixGroups(user,
+        (String u) -> ShellUtils.execCommand(ShellUtils.getGroupsForUserCommand(u)));
+  }
+
+  /**
+   * Gets the current user's group list from Unix by running the command 'groups' NOTE. For
+   * non-existing user it will return EMPTY list. This method may return duplicate groups.
+   *
+   * @param user user name
+   * @param groupsProvider function which given a username, returns a list of groups delimited by
+   *                       spaces
+   * @return the groups list that the {@code user} belongs to. The primary group is returned first
+   */
+  public static List<String> getUnixGroups(String user, IOFunction<String, String> groupsProvider)
```

Comment:
why do we need this method now?

Reply:
this is to avoid static mocks (powermock) in the tests. It lets us inject a mock group mapping provider for the unit tests